src/beamlime/logging/loggers.py

```diff
@@ -0,0 +1,127 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+import logging
+import sys
+from typing import Any, Callable, Union
+
+
+def hold_logging(logging_func: Callable) -> Callable:
```

Comment:
What does this function do?

Reply:
I locks the thread and it's often used in many places in ``logging`` so I thought it'd be better to have it as a decorator since both of them must be called before and after the function call.